stockflux-securities-master/src/components/InputForm.js

```diff
@@ -167,21 +190,34 @@ const InputForm = ({ match }) => {
               onChange={event => setEnabled(event.target.checked)}
             />
           </div>
-          <div className="input-submit-button-container">
-            <div
+          <div className="input-buttons-container">
+            <Button
+              size="large"
+              text={match.params.securityId ? "Save" : "Create"}
               className={
-                (formState === stateEnum.sending ? "in-progress " : "") +
+                (state === InputFormState.sending ? "in-progress " : "") +
                 "input-submit-button"
               }
-            >
-              <button>{match.params.securityId ? "Save" : "Create"}</button>
-            </div>
+            />
+            {match.params.securityId && (
+              <Confirmation confirmationText="Are you sure you want to delete this security?">
+                <Button
+                  type="button"
+                  className="input-delete-button"
+                  text="Delete"
+                  size="large"
+                  onClick={onClickDelete}
+                />
+              </Confirmation>
+            )}
+            {renderRedirect()}
```

Comment:
As mentioned above, I think you should conditionally render the Redirect component directly here. You can use the `redirect && <Redirect ... />` syntax to do it quite easily. To make it even more clear, perhaps add a conditional return quite near the top of your component function, before even bothering to return JSX:

    if (redirect) {
        return <Redirect ... />
    }

    // ...

Reply:
I've chosen the && method and kept it near the button as I think it's nice to keep it close to the button that renders it. They  did this in a tutorial I read as well